nex-rs/src/server.rs

```diff
@@ -17,39 +19,49 @@ use tokio::sync::Mutex;
 use tokio::task::JoinHandle;
 use tokio::time;
 
+#[derive(Debug, PartialEq, Snafu)]
+pub enum Error {
+    #[snafu()]
+    NoSocket,
+    #[snafu()]
+    CouldNoBindToAddress,
+    #[snafu()]
+    DataReceiveError,
+    #[snafu()]
+    DataSendError,
+    #[snafu(display(
+        "Tried to send too many fragments to {}: sequence_id 0x{:02x}, fragment_id 0x{:x}",
+        client_addr,
+        sequence_id,
+        fragment_id,
+    ))]
+    TooManyFragments {
+        client_addr: SocketAddr,
+        sequence_id: u16,
+        fragment_id: usize,
+    },
+}
+
+pub type ServerResult<T> = Result<T, Error>;
```

Comment:
Is this used? Looks like all the functions return a NexResult

Reply:
Good catch.  I meant to use `ServerResult` for most server methods, but looks like I used `NexResult` during my mass replace.